### ISSUE TITLE
[FR] Hooks

### ISSUE THREAD

**Issue by MonAaraj** [srevinsaju/zap#37]
Hooks would work really well in extending this package manager and would make any wanted features relatively easy to implement just given hooks, look at something like pacman's hook system

**Comment by srevinsaju** [comment]
Good idea, but a question.. where will we store the hooks? inside the zap github repo, or should we store them outside zap source code, in a separate github repo?

**Comment by MonAaraj** [comment]
I'd prefer in the same repo, maybe also have them be set up in `${XDG_DATA_HOME:-$HOME/.local/share}/zap/hooks`, but storing them seperately doesn't seem like a very bad idea.
